template/src/VideoSample/index.tsx

```diff
@@ -0,0 +1,130 @@
+import React, {useCallback, useEffect, useMemo} from 'react';
+import * as RN from 'react-native';
+import {Types, Fonts, Theme} from '@sleeperhq/mini-core';
+import {FlashList} from '@shopify/flash-list';
+import {Topic} from '@sleeperhq/mini-core/declarations/types';
+import VideoRow from './VideoRow';
+
+type VideoMiniProps = {
+  context: Types.Context;
+};
+
+const PAGE_LIMIT = 20;
+const EMPTY_OBJECT = {};
+
+const Videos = (props: VideoMiniProps) => {
+  const {context} = props;
+
+  const [page, setPage] = React.useState(0);
+  const [isRefreshing, setIsRefreshing] = React.useState(false);
+  const [paginatedVideosMap, setPaginatedVideoMap] =
+    React.useState<Record<string, Topic[]>>(EMPTY_OBJECT);
+
+  const currentPageVideos = context?.videos?.[page] || [];
+
+  useEffect(() => {
+    setIsRefreshing(false);
+    if (currentPageVideos.length !== 0) {
+      setPaginatedVideoMap({...paginatedVideosMap, [page]: currentPageVideos});
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [page, currentPageVideos, isRefreshing]);
+
+  const videos = useMemo(() => {
+    const videoArray = [];
+    Object.keys(paginatedVideosMap).forEach(videoIndex => {
+      const videoValues = paginatedVideosMap[videoIndex];
+      videoArray.push(...videoValues);
+    });
+    return videoArray;
+  }, [paginatedVideosMap]);
+
+  const onEndReached = useCallback(() => {
+    if (currentPageVideos.length === PAGE_LIMIT && isRefreshing === false) {
+      setPage(page + 1);
+    }
+  }, [isRefreshing, page, currentPageVideos.length]);
+
+  const onRefresh = useCallback(() => {
+    setIsRefreshing(true);
+    setPage(0);
+    setPaginatedVideoMap(EMPTY_OBJECT);
```

Comment:
This causes the entire list to go blank for a frame as is. I think if you merge this all together in `useMergeState` that issue would be fixed.